src/services/jetstreamService.ts

```diff
@@ -143,33 +143,43 @@ async function validateEnvironment(): Promise<void> {
 }
 
 async function createConsumer(functionName: string, jsm: JetStreamManager, consumerStreamName: string): Promise<void> {
-  await createStream(jsm, consumerStreamName);
+  await createStream(jsm, consumerStreamName, startupConfig.streamSubject ? startupConfig.streamSubject : undefined);
+  const streamSubjects = startupConfig.streamSubject ? startupConfig.streamSubject.split(',') : [startupConfig.consumerStreamName];
+
   const typedAckPolicy = startupConfig.ackPolicy;
   const consumerCfg: Partial<ConsumerConfig> = {
     ack_policy: AckPolicy[typedAckPolicy],
     durable_name: functionName,
+    filter_subjects: streamSubjects,
   };
   await jsm.consumers.add(consumerStreamName, consumerCfg);
   logger.log('Connected Consumer to Consumer Stream');
 }
 
-async function createStream(jsm: JetStreamManager, streamName: string): Promise<void> {
+async function createStream(jsm: JetStreamManager, streamName: string, subjectName?: string): Promise<void> {
   await jsm.streams.find(streamName).then(
-    (s) => {
-      logger.log('Producer stream already exists');
+    async (stream) => {
+      logger.log(`Stream: ${streamName} already exists.`);
+
+      if (subjectName) {
+        logger.log(`Adding subject: ${subjectName} to stream: ${streamName}`);
+        const streamInfo = await jsm.streams.info(stream);
+        streamInfo.config.subjects?.push(subjectName);
```

Comment:
If there is no subjects, this push will not add any subjects? So don't you need an if check to say if no subjects, create?